modules/drone_odometry_local.py

```diff
@@ -0,0 +1,124 @@
+"""
+Drone odometry in local space (origin at home location).
+"""
+
+from .common.mavlink.modules import drone_odometry
+
+
+# Basically a struct
+# pylint: disable=too-few-public-methods
+class DronePositionLocal:
+    """
+    Drone position in NED system.
+    """
+    __create_key = object()
+
+    @classmethod
+    def create(cls,
+               north: float,
+               east: float,
+               down: float) -> "tuple[bool, DronePositionLocal | None]":
+        """
+        north, east, down in metres.
+        """
+        # Cannot be underground
+        if down >= 0.0:
+            return False, None
+
+        return True, DronePositionLocal(cls.__create_key, north, east, down)
+
+    def __init__(self,
+                 class_private_create_key,
+                 north: float,
+                 east: float,
+                 down: float):
+        """
+        Private constructor, use create() method.
+        """
+        assert class_private_create_key is DronePositionLocal.__create_key, "Use create() method"
+
+        self.north = north
+        self.east = east
+        self.down = down
+
+# pylint: enable=too-few-public-methods
+
+
+# Basically a struct
+# pylint: disable=too-few-public-methods
+class DroneOrientationLocal:
+    """
+    Wrapper for DroneOrientation as it is the same in both local and global space.
+    """
+    __create_key = object()
+
+    @classmethod
+    def create_new(cls,
+                   yaw: float,
+                   pitch: float,
+                   roll: float) -> "tuple[bool, DroneOrientationLocal | None]":
+        """
+        yaw, pitch, roll in radians.
+        """
+        result, orientation = drone_odometry.DroneOrientation.create(yaw, pitch, roll)
+        if not result:
+            return False, None
+
+        # Get Pylance to stop complaining
+        assert orientation is not None
+
+        return True, DroneOrientationLocal(cls.__create_key, orientation)
+
+    @classmethod
+    def create_wrap(cls,
```

Comment:
why do we need to create a wrap?

Reply:
Rotation is independent from translation, and code duplication so composition used instead. But we want to avoid making the caller create a `DroneOrientationLocal` , then check if it succeeded, then attempt to create. We also want to avoid dropping and recreating the object so this method exists. Unfortunately function overloading does not exist (I tried :( ).

Reply:
No way to use kwargs or something to "pseudo overload" functions?

Reply:
kwargs avoided as the type is not enforced by a linter (the only "compile-time protection" Python may have).